logstash-core/src/main/java/org/logstash/Logstash.java

```diff
@@ -0,0 +1,144 @@
+package org.logstash;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jruby.Ruby;
+import org.jruby.RubyException;
+import org.jruby.RubyInstanceConfig;
+import org.jruby.RubyNumeric;
+import org.jruby.exceptions.RaiseException;
+import org.jruby.runtime.builtin.IRubyObject;
+
+/**
+ * Logstash Main Entrypoint.
+ */
+public final class Logstash implements Runnable, AutoCloseable {
+
+    private static final Logger LOGGER = LogManager.getLogger(Logstash.class);
+
+    /**
+     * Configuration for {@link #ruby}.
+     */
+    private final RubyInstanceConfig config;
+
+    /**
+     * JRuby Runtime Environment.
+     */
+    private final Ruby ruby;
+
+    /**
+     * Ruby Entrypoint Script.
+     */
+    private final InputStream script;
+
+    /**
+     * Main Entrypoint.
+     * Requires environment {@code "LS_HOME"} to be set to the Logstash root directory.
+     * @param args Logstash CLI Arguments
+     */
+    public static void main(final String... args) {
+        final String lsHome = System.getenv("LS_HOME");
+        if (lsHome == null) {
+            throw new IllegalStateException("LS_HOME environment variable must be set.");
+        }
+        final Path home = Paths.get(lsHome).toAbsolutePath();
```

Comment:
Can we verify some properties about this path before using it: Does the path exist? Does it have things we want in it? etc..

My thought is we can do a quick check here and provide actionable response to the user if the check fails.

Reply:
@jordansissel same as the above, this will actually throw something like "file not found" if the path is wrong which we will understand, for the user this means there is a bug in the startup script in 100% of cases imo => nothing actionable they can do right?